suite-common/wallet-core/src/discovery/discoveryThunks.ts

```diff
@@ -430,13 +433,12 @@ export const getAvailableCardanoDerivationsThunk = createThunk(
 
 export const startDiscoveryThunk = createThunk(
     `${DISCOVERY_MODULE_PREFIX}/start`,
-    async (_, { dispatch, getState, extra }): Promise<void> => {
+    async (device: TrezorDevice, { dispatch, getState, extra }): Promise<void> => {
```

Comment:
here, bind the discovery always to the passed device NOT to the selected device.